doc/ioc/tutorials/streamdevice.md

```diff
@@ -0,0 +1,288 @@
+---
+title: "Creating a StreamDevice IOC"
+---
+
+In this tutorial,
+you're gonna learn how to create an EPICS IOC with EPNix
+that communicates with a power supply,
+using the [StreamDevice] support module.
+
+  [StreamDevice]: https://paulscherrerinstitute.github.io/StreamDevice/
+
+# Pre-requisites
+
+Verify that you have all pre-requisites installed.
+If not,
+follow the [Pre-requisites] section.
+
+  [Pre-requisites]: ./pre-requisites.md
+
+# Running the power supply simulator
+
+EPNix has a power supply simulator
+for you to test your IOC.
+
+To run it:
+
+``` bash
+nix run 'github:epics-extensions/epnix#psu-simulator'
+```
+
+For the rest of the tutorial,
+leave it running in a separate terminal.
+
+# Creating your top
+
+We can use these command to create an EPNix top:
+
+``` bash
+# Initialise an EPNix top
+nix flake new -t 'github:epics-extensions/epnix' my-top
+cd my-top
+
+# Enter the EPNix development shell, that has EPICS base installed in it.
+nix develop
+
+# Create your app and ioc boot folder
+makeBaseApp.pl -t ioc example
+makeBaseApp.pl -i -t ioc -p example -a linux-x86_64 Example
+
+# Create a git repository, and make sure all files are tracked
+git init
+git add .
+```
+
+After that,
+you can already check that your top build with:
+
+``` bash
+nix build -L
+```
+
+This `nix build`{.sh} command compiles your IOC,
+and all its dependencies.
+This makes the usual EPICS environment setup unneeded.
+
+If found in the official Nix cache server,
+Nix downloads packages from there
+instead of compiling them.
+
+This command puts a `./result` symbolic link in your current directory,
+containing the compilation result.
+
+# Adding StreamDevice to the EPNix environment
+
+Adding dependencies to the EPNix environment happen inside the `flake.nix` file.
+This file is the main entry point for specifying your build environment:
+most Nix commands used here read this file to work.
+
+For adding StreamDevice,
+change yours like so:
+
+``` {.diff filename="flake.nix"}
+         # Add one of the supported modules here:
+         # ---
+-        #support.modules = with pkgs.epnix.support; [ StreamDevice ];
++        support.modules = with pkgs.epnix.support; [ StreamDevice ];
+```
+
+Then,
+leave your EPNix development shell by running `exit`{.sh},
```

Comment:
I'm not sure I understand the question, but this line also provides StreamDevice in the development shell. So you have to leave and re-enter to make sure your environment contains StreamDevice. `nix flake update` is only about flake inputs.

Reply:
it was my question, I hoped `nix flake update` was for any modification in `flake.nix` not just only for the inputs.

Is there any  `nix develop restart` ? Just to avoid to do 2 commands. 

Quite minor issue, you can close this thread if no

Reply:
> Is there any nix develop restart ? Just to avoid to do 2 commands.

That's not possible today. You might be able to use `direnv` with the Nix flakes integration, which does everything automatically, but this is another tool to learn and to install, so I'm not mentioning it.